File: javascript/콜라츠 추측.js
function solution(num) {
    
    let cnt = 0;
    
    while (true){
        if (num === 1) { return cnt;}
        
        if (num%2 === 0) {
            num /= 2;
            cnt++;
        } else if (num%2 === 1) {
            num = num*3+1;
            cnt++;
        }

        if (cnt === 500) { break;}
    }
    return -1;
}

File: java/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        if(num==1) return 0;
        
        for(int i=0; i<500; i++){
            if(i==499) return -1;
            
            if(num%2==0) num /=2;
            else if(num%2==1) num = num*3+1;
            
            if(num==1) return i+1; 
        }
        return 0;
        
    }
}
